### PR TITLE
Refactor the WasmEdge Book.

### PR DESCRIPTION
**Pull request by q82419**
@hydai , @juntao , @alabulei1 
Please check the table of contents. (Can refer to the SUMMARY.md)

- [x] Introduction
- [x] Quick Start
- [x] WasmEdge Features
- [x] WasmEdge Use Cases
- [x] Write a WebAssembly Application
- [x] Use WasmEdge Library
- [x] Use WasmEdge CLI
- [x] Develop WasmEdge Plug-in
- [x] Contribute to WasmEdge (WIP)

### CODE REVIEW

**Comment by q82419** [comment]
@juntao , @alabulei1 

Please help to resolve the lost link:

use_cases/kubernetes/kubernetes/superedge.md:5
use_cases/kubernetes/kubernetes/superedge.md:27
use_cases/kubernetes/kubernetes/superedge.md:29

**Comment by q82419** [comment]
@juntao , @alabulei1 , @apepkuss 
Please help to resove the following issue:

use_cases/js_or_dsl_runtime.md: line 10-13
(May exists in book? Change the link into book)

sdk/node.md: line 10
(The rustwasmc is removed in book, need to update to use wasmedge_bindgen in examples on this page)

sdk/go/bindgen.md:
(The rustwasmc is removed in book, need to update to use wasmedge_bindgen in examples on this page)

**Comment by apepkuss** [comment]
Hi @DarumaDocker,

@q82419 mentioned the following two markdown issues that are related to `wasmedge_bindgen`. I guess that could be part of your demo examples. Could you please help check them? Thanks!

sdk/node.md: line 10
(The rustwasmc is removed in book, need to update to use wasmedge_bindgen in examples on this page)

sdk/go/bindgen.md:
(The rustwasmc is removed in book, need to update to use wasmedge_bindgen in examples on this page)

**Comment by q82419** [comment]
@alabulei1 
The followings are the link changing:
(Links have `https://wasmedge.org/book/en/` prefix)


intro.html -> index.html

start/install.html -> quick_start/install.html
start/docker.html -> quick_start/use_docker.html
start/cli.html -> quick_start/run_cli.html
start/universal.html -> quick_start/run_in_aot_mode.html

intro/features.html -> features.html
intro/standard.html -> features/proposals.html
intro/proprietary.html -> features/proprietary_extend.html
intro/integrations.html -> features/integrations.html
os.html -> features/platforms.html
intro/faq.html -> features/comparison.html

intro/use.html -> use_cases.html
intro/use/jamstack.html -> use_cases/server_side_render.html
intro/use/microservice.html -> use_cases/microservice.html
intro/use/serverless.html -> use_cases/serverless_faas.html
intro/use/saas.html -> use_cases/serverless_saas.html
intro/use/device.html -> use_cases/smart_device.html
intro/use/js.html -> use_cases/js_or_dsl_runtime.html

kubernetes.html -> use_cases/kubernetes.html
kubernetes.png -> use_cases/kubernetes.png
kubernetes/quickstart.html -> use_cases/kubernetes/quickstart.html
kubernetes/demo.html -> use_cases/kubernetes/demo.html
kubernetes/demo/wasi.html -> use_cases/kubernetes/demo/wasi.html
kubernetes/demo/server.html -> use_cases/kubernetes/demo/server.html
kubernetes/container.html -> use_cases/kubernetes/container.html
kubernetes/container/crun.html -> use_cases/kubernetes/container/crun.html
kubernetes/container/runc.html -> use_cases/kubernetes/container/runc.html
kubernetes/container/youki.html -> use_cases/kubernetes/container/youki.html
kubernetes/cri.html -> use_cases/kubernetes/cri.html
kubernetes/cri/crio.html -> use_cases/kubernetes/cri/crio.html
kubernetes/cri/containerd.html -> use_cases/kubernetes/cri/containerd.html
kubernetes/kubernetes.html -> use_cases/kubernetes/kubernetes.html
kubernetes/kubernetes/kubernetes-crio.html -> use_cases/kubernetes/kubernetes/kubernetes-crio.html
kubernetes/kubernetes/kubernetes-containerd.html -> use_cases/kubernetes/kubernetes/kubernetes-containerd.html
kubernetes/kubernetes/kind.html -> use_cases/kubernetes/kubernetes/kind.html
kubernetes/kubernetes/kubeedge.html -> use_cases/kubernetes/kubernetes/kubeedge.html
kubernetes/kubernetes/superedge.html -> use_cases/kubernetes/kubernetes/superedge.html
kubernetes/kubernetes/openyurt.html -> use_cases/kubernetes/kubernetes/openyurt.html
kubernetes/docker.html -> use_cases/kubernetes/docker.html
kubernetes/docker/lxc.html -> use_cases/kubernetes/docker/lxc.html
kubernetes/docker/containerd.html -> use_cases/kubernetes/docker/containerd.html

frameworks.html -> use_cases/frameworks.html
frameworks/mesh.html -> use_cases/frameworks/mesh.html
frameworks/mesh/dapr.html -> use_cases/frameworks/mesh/dapr.html
frameworks/mesh/mosn.html -> use_cases/frameworks/mesh/mosn.html
frameworks/mesh/wasm-nginx-module.html -> use_cases/frameworks/mesh/wasm-nginx-module.html
frameworks/mesh/eventmesh.html -> use_cases/frameworks/mesh/eventmesh.html
frameworks/app.html -> use_cases/frameworks/app.html
frameworks/app/yomo.html -> use_cases/frameworks/app/yomo.html
frameworks/app/reactr.html -> use_cases/frameworks/app/reactr.html
frameworks/serverless.html -> use_cases/frameworks/serverless.html
frameworks/serverless-wasmedge.png -> use_cases/frameworks/serverless-wasmedge.png
frameworks/serverless/vercel.html -> use_cases/frameworks/serverless/vercel.html
frameworks/serverless/netlify.html -> use_cases/frameworks/serverless/netlify.html
frameworks/serverless/aws.html -> use_cases/frameworks/serverless/aws.html
frameworks/serverless/tencent.html -> use_cases/frameworks/serverless/tencent.html
frameworks/serverless/secondstate.html -> use_cases/frameworks/serverless/secondstate.html

dev.html -> write_wasm.html
dev/rust.html -> write_wasm/rust.html
dev/rust/bindgen.html -> write_wasm/rust/bindgen.html
dev/rust/command.html -> write_wasm/rust/command.html
dev/rust/networking.html -> write_wasm/rust/networking.html
dev/rust/networking-https.html -> write_wasm/rust/networking-https.html
dev/rust/networking-nonblocking.html -> write_wasm/rust/networking-nonblocking.html
dev/rust/ssr.html -> write_wasm/rust/ssr.html
dev/rust/tensorflow.html -> write_wasm/rust/tensorflow.html
dev/rust/wasi.html -> write_wasm/rust/wasi.html
dev/rust/wasicrypto.html -> write_wasm/rust/wasicrypto.html
dev/rust/wasinn.html -> write_wasm/rust/wasinn.html
dev/js.html -> write_wasm/js.html
dev/js/es6.html -> write_wasm/js/es6.html
dev/js/modules.html -> write_wasm/js/modules.html
dev/js/networking.html -> write_wasm/js/networking.html
dev/js/nodejs.html -> write_wasm/js/nodejs.html
dev/js/npm.html -> write_wasm/js/npm.html
dev/js/quickstart.html -> write_wasm/js/quickstart.html
dev/js/rust.html -> write_wasm/js/rust.html
dev/js/ssr.html -> write_wasm/js/ssr.html
dev/js/tensorflow.html -> write_wasm/js/tensorflow.html
dev/as.html -> write_wasm/as.html
dev/grain.html -> write_wasm/grain.html
dev/go.html -> write_wasm/tinygo.html
dev/kotlin.html -> write_wasm/kotlin.html
dev/python.html -> write_wasm/python.html
dev/swift.html -> write_wasm/swift.html

embed.html -> sdk.html
embed/c.html -> sdk/c.html
extend/plugin/externref.html -> sdk/c/externref.html
extend/plugin/hostfunction.html -> sdk/c/hostfunction.html
embed/c/ref.html -> sdk/c/ref.html
embed/c/0.10.1/ref.html -> sdk/c/0.10.1/ref.html
embed/c/0.10.1/upgrade_to_0.11.0.html -> sdk/c/0.10.1/upgrade_to_0.11.0.html
embed/c/0.9.1/ref.html -> sdk/c/0.9.1/ref.html
embed/c/0.9.1/upgrade_to_0.10.0.html -> sdk/c/0.9.1/upgrade_to_0.10.0.html
embed/go.html -> sdk/go.html
embed/go/ref.html -> sdk/go/ref.html
embed/go/app.html -> sdk/go/app.html
embed/go/bindgen.html -> sdk/go/bindgen.html
embed/go/function.html -> sdk/go/function.html
embed/go/memory.html -> sdk/go/memory.html
embed/go/tensorflow.html -> sdk/go/tensorflow.html
embed/go/0.9.1/ref.html -> sdk/go/0.9.1/ref.html
embed/go/0.9.1/upgrade_to_0.10.0.html -> sdk/go/0.9.1/upgrade_to_0.10.0.html
embed/node.html -> sdk/node.html
embed/python.html -> sdk/python.html
embed/rust.html -> sdk/rust.html
embed/rust/concurrent_fib.html -> sdk/rust/concurrent_fib.html
embed/rust/how_to_use_module_instance.html -> sdk/rust/how_to_use_module_instance.html
embed/rust/memory_manipulation.html -> sdk/rust/memory_manipulation.html
embed/rust/say_hello.html -> sdk/rust/say_hello.html
embed/rust/sys_run_host_func.html -> sdk/rust/sys_run_host_func.html
embed/rust/table_and_funcref.html -> sdk/rust/table_and_funcref.html

extend/plugin.html -> plugin.html
extend/plugin/loadable.html -> plugin.html

(`Contribute` WIP)

The followings are the link changes of the old reprecated docs outside of the book:

https://github.com/WasmEdge/WasmEdge/blob/master/docs/CONTRIBUTING.md -> https://wasmedge.org/book/en/contribute.html
https://github.com/WasmEdge/WasmEdge/blob/master/docs/build.md -> https://wasmedge.org/book/en/contribute/build_from_src.html
https://github.com/WasmEdge/WasmEdge/blob/master/docs/build_on_mac.md -> https://wasmedge.org/book/en/contribute/build_from_src/macos.html
https://github.com/WasmEdge/WasmEdge/blob/master/docs/build_on_windows.md -> https://wasmedge.org/book/en/contribute/build_from_src/windows.html
https://github.com/WasmEdge/WasmEdge/blob/master/docs/c_api.md -> https://wasmedge.org/book/en/sdk/c/ref.html
https://github.com/WasmEdge/WasmEdge/blob/master/docs/c_api_quick_start.md -> https://wasmedge.org/book/en/sdk/c.html
https://github.com/WasmEdge/WasmEdge/blob/master/docs/extensions.md -> https://wasmedge.org/book/en/features/proprietary_extend.html
https://github.com/WasmEdge/WasmEdge/blob/master/docs/externref.md -> https://wasmedge.org/book/en/sdk/c/externref.html
https://github.com/WasmEdge/WasmEdge/blob/master/docs/go_api.md -> https://wasmedge.org/book/en/sdk/go/ref.html
https://github.com/WasmEdge/WasmEdge/blob/master/docs/highlights.md -> https://wasmedge.org/book/en/features.html
https://github.com/WasmEdge/WasmEdge/blob/master/docs/host_function.md -> https://wasmedge.org/book/en/plugin/hostfunction.html
https://github.com/WasmEdge/WasmEdge/blob/master/docs/install.md -> https://wasmedge.org/book/en/quick_start/install.html
https://github.com/WasmEdge/WasmEdge/blob/master/docs/integrations.md -> https://wasmedge.org/book/en/features/integrations.html
https://github.com/WasmEdge/WasmEdge/blob/master/docs/run_javascript.md -> https://wasmedge.org/book/en/quick_start/run_cli.html
https://github.com/WasmEdge/WasmEdge/blob/master/docs/run.md -> https://wasmedge.org/book/en/cli.html
https://github.com/WasmEdge/WasmEdge/blob/master/docs/use_cases.md -> https://wasmedge.org/book/en/use_cases.html
https://github.com/WasmEdge/WasmEdge/blob/master/docs/showcase.md -> https://wasmedge.org/book/en/use_cases.html
https://github.com/WasmEdge/WasmEdge/blob/master/docs/simd.md -> https://wasmedge.org/book/en/write_wasm/c/simd.html

**Review by dm4 — APPROVED** [review]
Pretty GOOD!!!! 👍👍👍

**Review by dm4 — CHANGES_REQUESTED** [review]
As discussed in meeting, please add redirects for existed urls.

Ref: https://rust-lang.github.io/mdBook/format/configuration/renderers.html#outputhtmlredirect

File: docs/book/en/book.toml [hunk 1]
```diff
@@ -31,5 +31,135 @@ expand = true
...
+"/dev/js/networking.html" = "/write_wasm/js/networking.html"
+"/dev/js/nodejs.html" = "/write_wasm/js/nodejs.html"
+"/dev/js/npm.html" = "/write_wasm/js/npm.html"
+"/dev/js/quickstart.html" = "/write_wasm/js/quickstart.html"
+"/dev/js/rust.html" = "/write_wasm/js/rust.html"
+"/dev/js/ssr.html" = "/write_wasm/js/ssr.html"
+"/dev/js/tensorflow.html" = "/write_wasm/js/tensorflow.html"
+"/dev/as.html" = "/write_wasm/as.html"
+"/dev/grain.html" = "/write_wasm/grain.html"
+"/dev/go.html" = "/write_wasm/tinygo.html"
```
Comment (dm4) [inline docs/book/en/book.toml:113]: Should be `/write_wasm/go.html`?

**Review by dm4 — CHANGES_REQUESTED** [review]

**Review by dm4 — APPROVED** [review]
Pretty GOOD!!!! 👍👍👍 (again)

**Pull request merged**